habitat_baselines/models/rcm/vln_rcm_policy.py

```diff
@@ -270,7 +270,12 @@ def forward(self, observations, rnn_hidden_states, prev_actions, masks):
 
         if self.vln_config.PROGRESS_MONITOR.use and AuxLosses.is_active():
             progress_hat = torch.tanh(self.progress_monitor(x))
-            progress_loss = F.mse_loss(progress_hat.squeeze(1), observations['progress'])
+            progress_loss = F.mse_loss(
+                progress_hat.squeeze(1),
+                observations['progress'],
+                reduction='none',
+            )
+            progress_loss = (progress_loss.unsqueeze(1) * masks).sum() / masks.sum()
```

Comment:
Unfortunately these masks are not are about padding, they are episodes starting/ending (masks[t, n] == 0 denotes an episode beginning at that step).  We could use them for padding also, with the caveat that masks[0, :] will be zero instead of 1, but we would need to wire them up to do that (right now masks[1:] == 1.0 regardless of padding).

Probably the correct thing to do is give AuxLosses a (t, n) (or (t*n,)) tensor here and then extend the `reduce()` method to also take a mask.

Reply:
I fixed this for now using `weights > 0` in the `_update_agent` call.

I also used losses and masks with shape (t*n). I think I like (t, n) better so I might go back and change that at some point.